src/encoded/static/components/dataset.js

```diff
@@ -1894,10 +1894,12 @@ export const SeriesComponent = (props, reactContext) => {
                                 </div>
                             : null}
 
-                            <div data-test="donordiversity">
-                                <dt>Donor diversity</dt>
-                                <dd>{diversity}</dd>
-                            </div>
+                            {seriesType !== 'SingleCellRnaSeries' ?
+                                <div data-test="donordiversity">
+                                    <dt>Donor diversity</dt>
+                                    <dd>{diversity}</dd>
+                                </div>
+                            : null}
```

Comment:
While this is correct, I think it would be less prone to future breakage if you used the `diversity` variable as the condition for displaying itself so that you test the very thing you display, e.g.:

```
{diversity ?
    <div data-test="donordiversity">
        <dt>Donor diversity</dt>
        <dd>{diversity}</dd>
     </div>
: null}
```
…kind of thing. Then above on line 1836 you can do something like:
```
    const diversity = seriesType !== 'SingleCellRnaSeries' ? donorDiversity(context) : null;
```
That way if some future condition includes no diversity, you can that logic where we calculate the diversity, and the part where you display `diversity` remains unchanged. Plus the `donorDiversity` calculation is a little bit involved, so it’s nice to skip it completely if you don’t need it.

Reply:
Thank you @forresttanaka! That makes a lot of sense (especially avoiding the calculation altogether if you don't need to do it), I have a lot to learn when it comes to JS. Will try implementing your suggestion.